app/src/main/java/com/gatchimath/totask/SqlStorage.java

```diff
@@ -0,0 +1,88 @@
+package com.gatchimath.totask;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import java.util.ArrayList;
+
+/**
+ * Represents all task-related SQL storage.
+ * <p>
+ * Most of the methods here are basically just SQLite commands.
+ * This class should only be instantiated in a
+ * {@link com.gatchimath.totask.TaskStorage}.
+ *
+ * @see com.gatchimath.totask.TaskTableEntry,
+ *      com.gatchimath.totask.TaskDbHelper
+ */
+public class SqlStorage {
+	
+	private TaskDbHelper dbHelper;
+	
+	public SqlStorage(Context context) {
+		dbHelper = new TaskDbHelper(context);
+	}
+	
+	/**
+	 * Writes the task to the task table.
+	 * More info relating to the task table can be found in the
+	 * {@link com.gatchimath.totask.TaskTableEntry} and
+	 * {@link com.gatchimath.totask.TaskDbHelper} classes.
+	 *
+	 * @param task  Task to write to the table.
+	 */
+	public void writeTask(Task task) {
+		SQLiteDatabase db = dbHelper.getWritableDatabase();
+		ContentValues values = new ContentValues();
+		
+		values.put(TaskTableEntry.COLUMN_NAME_1, task.getName());
+		values.put(TaskTableEntry.COLUMN_NAME_2, task.getDetails());
+		values.put(TaskTableEntry.COLUMN_NAME_3, task.isDone() ? 1 : 0);
+		db.insert(TaskTableEntry.TABLE_NAME, null, values);
+	}
+	
+	/**
+	 * Returns a list of all the tasks in the table.
+	 * @return  ArrayList of Tasks in the table.
+	 */
+	public ArrayList<Task> retrieveAllTasks() {
+		SQLiteDatabase db = dbHelper.getReadableDatabase();
+		Cursor cursor = db.query(
+			TaskTableEntry.TABLE_NAME,
+			null, null,	null, null,	null, null
+		);
+		
+		// Now exract the info from the cursor
+		cursor.moveToFirst();
```

Comment:
This may be unnecessary. You should be able to check easily once you have unit tests by deleting it and seeing if they still pass.